h/api/search/query.py

```diff
@@ -68,25 +88,25 @@ def build(request_params, userid=None):
 
     for key, value in request_params.items():
         matches.append({"match": {key: value}})
-    matches = matches or [{"match_all": {}}]
 
-    query["query"] = {"bool": {"must": matches}}
+    # Add a filter for "not in public site areas" considerations
+    filters.append(nipsa.nipsa_filter(userid=userid))
 
-    query["query"] = {
-        "filtered": {
-            "filter": nipsa.nipsa_filter(userid=userid),
-            "query": query["query"]
-        }
-    }
+    # And now we simplify the query. If there are no added matches, then we can
```

Comment:
Or, alternatively, let's simplify this function a bit and make it clearer. Start with explicit empty lists:

```python
filter_clauses = []
query_clauses = []
```

Then append as necessary.

Finally, construct the whole body rather than mutating tit.